chia/simulator/block_tools.py

```diff
@@ -403,7 +403,10 @@ def setup_new_gen(
             assert rng is not None
             bundle, additions = make_spend_bundle(available_coins, wallet, rng)
             removals = bundle.removals()
-            program = simple_solution_generator(bundle).program
+            if curr.height >= self.constants.HARD_FORK_HEIGHT:
```

Comment:
this is a reasonable change. It doesn't affect the number of transactions per block (which is good, that should stay the same). It does, however, change the chain, so the `test_cache` would have to be updated. Doing so expands the scope of this task quite a lot, so I think it probably best to avoid it.